### PR TITLE
(Enhancement) Adds torrents/edit API endpoint

### PR DESCRIPTION
**Pull request by PyR8zdl**
This adds torrents/edit API endpoint.
It will edit the torrents data if param is passed and the user is owner.

### CODE REVIEW

**Comment by PyR8zdl** [comment]
I still want to add an option to delete the torrent. Still working on the best way to do this.

File: app/Http/Controllers/API/TorrentController.php [hunk 1]
```diff
@@ -241,6 +278,70 @@ public function show($id)
...
+     * @param int     $id
+     *
+     * @return TorrentResource
+     */
+    public function edit(Request $request, $id)
+    {
+        $user = $request->user();
+        if ($user->group->is_owner) {
+            $torrent = Torrent::findOrFail($id);
+            if ($request->filled('name')) {
```
Comment (PyR8zdl) [inline app/Http/Controllers/API/TorrentController.php]: @HDVinnie For these if statements I only want to set $torrent->x if $request->filled('x') with no else. I don't want to touch the data if the request is supplied.

I found this stackoverflow, which one should I adopt to? I know PHP7.4 will be required soon but in the meantime I don't want to break anything for others who haven't updated.
https://stackoverflow.com/a/36933323/12882632

**Comment by PyR8zdl** [comment]
I'm working on torrent delete in API however I will not be able to test till Wednesday.
It's mostly a copy-paste from `/app/Http/Controllers/TorrentController.php @deleteTorrent`

Then I will start work on user API endpoint.

File: app/Http/Controllers/API/TorrentController.php (hunk 1 above)
Comment (HDVinnie) [reply app/Http/Controllers/API/TorrentController.php]: its fine how it is since these are only ifs and not if else.

**Pull request closed without merging**